packages/backend/src/graphql/mutations/update-flow-transfer-status.ts

```diff
@@ -1,24 +1,125 @@
 import { IFlowTransferStatus } from '@plumber/types'
 
+import logger from '@/helpers/logger'
+import Flow from '@/models/flow'
 import FlowTransfer from '@/models/flow-transfers'
+import Step from '@/models/step'
+import Context from '@/types/express/context'
 
 type Params = {
   input: {
     id: string
     status: IFlowTransferStatus
+    newOwnerId: string
   }
 }
 
-const updateFlowTransferStatus = async (_parent: unknown, params: Params) => {
-  const { id, status } = params.input
+/**
+ * Note that this mutation does two things
+ * 1. Cancel/reject a flow transfer
+ * 2. Approve a flow transfer AND update flow owner id as a single transaction
+ */
+
+const updateFlowTransferStatus = async (
+  _parent: unknown,
+  params: Params,
+  context: Context,
+) => {
+  // TODO (mal): check if need to stop possible exploits
+  const { id, status, newOwnerId } = params.input
+
+  if (status === 'pending') {
+    throw new Error('No updating of pipe transfer back to pending')
+  }
+
   const flowTransfer = await FlowTransfer.query()
     .findOne({
       id,
     })
+    .withGraphFetched({
+      newOwner: true,
+    })
     .throwIfNotFound()
 
-  return await flowTransfer.$query().patchAndFetch({
-    status,
+  // To prevent possible exploits: for approved/rejected status: check if new owner matches
+  if (
+    (status === 'approved' || status === 'rejected') &&
+    flowTransfer.newOwnerId !== newOwnerId
```

Comment:
Is there a reason you're not checking against currentUser? Never trust the client.

```suggestion
    flowTransfer.newOwnerId !== context.currentUser.id
```

Reply:
thanks for catching the mistake